js/api/auth/login.mjs

```diff
@@ -0,0 +1,31 @@
+import { API_SOCIAL_URL } from "../constants.mjs";
+import * as storage from "../../storage/index.mjs";
+
+const action = "/auth/login";
+const method = "post";
+
+export async function login(profile) {
+    const loginURL = API_SOCIAL_URL + action;
+    console.log(loginURL);
+
+    const body = JSON.stringify(profile);
+
+    const response = await fetch(loginURL, {
+    headers: {
+        "Content-Type": "application/json"
+    },
+    method,
+    body
+})
+
+const { accessToken, ...user } = await response.json();
+
+storage.save("accessToken", accessToken);
+
+storage.save("profile", user);
+
+alert("You have been logged in successfully");
```

Comment:
The error handling can be improved.

I'll just use this Fetch as the example but try to use Try-Catch blocks on your fetches. Something like: 

`export async function login(profile) {
    const loginURL = API_SOCIAL_URL + action;
    console.log(loginURL);

    const body = JSON.stringify(profile);

    const response = await fetch(loginURL, {
    headers: {
        "Content-Type": "application/json"
    },
    method,
    body
})

const { accessToken, ...user } = await response.json();

storage.save("accessToken", accessToken);

storage.save("profile", user);

alert("You have been logged in successfully");`